### PR TITLE
Added uniqueness validation for some Login fields

### PR DESCRIPTION
**Pull request by mathiasAichinger**
Solves #35

### CODE REVIEW

File: spec/models/login_spec.rb [hunk 1]
```diff
@@ -23,6 +23,30 @@
...
+    expect(login2).to_not be_valid
+  end
+
+  it 'validates uniqueness of single_use_oauth2_token' do
+    login = described_class.new(single_use_oauth2_token: 'oldtoken')
+    login.save
+    login2 = described_class.new(single_use_oauth2_token: 'oldtoken')
+
+    expect(login2).to_not be_valid
+  end
```
Comment (marcoow) [inline spec/models/login_spec.rb]: You can use the `validate_uniqueness_of` matcher here: http://www.rubydoc.info/gems/shoulda-matchers/Shoulda%2FMatchers%2FActiveRecord%3Avalidate_uniqueness_of

**Comment by marcoow** [comment]
cool - just one minor suggestion to clean up the tests a bit.

File: app/models/login.rb [hunk 2]
```diff
@@ -33,9 +33,9 @@ class InvalidOAuth2Token < StandardError; end
     has_secure_password
   end
 
-  validates :identification, presence: true
-  validates :oauth2_token, presence: true
-  validates :single_use_oauth2_token, presence: true
+  validates :identification, presence: true, uniqueness: true
+  validates :oauth2_token, presence: true, uniqueness: true
+  validates :single_use_oauth2_token, presence: true, uniqueness: true
```
Comment (mathiasAichinger) [inline app/models/login.rb]: `single_use_oauth2_token` is always generated right? so I couldn't figured how to test it

Comment (marcoow) [inline app/models/login.rb]: Did you try using the [shoulda matcher](http://www.rubydoc.info/gems/shoulda-matchers/Shoulda%2FMatchers%2FActiveRecord%3Avalidate_uniqueness_of) to test this?

Comment (mathiasAichinger) [inline app/models/login.rb]: yes and the uniqueness check failed

Comment (marcoow) [inline app/models/login.rb]: The only way to test that I could think of then is to `stub` the `ensure_oauth2_token` and `assign_single_use_oauth2_token` methods on the instance then to return a static value then. Stubbing private API isn't great but still better than not testing this at all.

**Comment by mathiasAichinger** [comment]
@marcoow I did the requested changes.

File: spec/models/login_spec.rb [hunk 3]
```diff
@@ -3,7 +3,19 @@
...
   end
 
-  it { is_expected.to validate_presence_of(:identification) }
+  subject { Login.new(identification: 'test@example.com', oauth2_token: 'token', single_use_oauth2_token: 'oldtoken') }
+
+  describe 'Login model validations' do
+    before do
+      Login.any_instance.stub(:ensure_oauth2_token).and_return(true)
+      Login.any_instance.stub(:assign_single_use_oauth2_token).and_return(true)
+    end
```
Comment (marcoow) [inline spec/models/login_spec.rb]: I think these mocks should enable adding specs for the presence validations of `oauth2_token` and `single_use_oauth2_token`? Can you add those below?

File: spec/models/login_spec.rb [hunk 4]
```diff
@@ -3,7 +3,19 @@
     expect(subject).to belong_to(:account).with_foreign_key(:user_id)
   end
 
-  it { is_expected.to validate_presence_of(:identification) }
+  subject { Login.new(identification: 'test@example.com', oauth2_token: 'token', single_use_oauth2_token: 'oldtoken') }
+
+  describe 'Login model validations' do
```
Comment (marcoow) [inline spec/models/login_spec.rb]: `Login model` is redundant here - I think only `describe 'validations'`is better.

**Comment by marcoow** [comment]
replaced by #45

**Pull request closed without merging**